src/tasks/TaskHelper.ts

```diff
@@ -253,3 +254,26 @@ async function findTaskByLabel(allTasks: TaskDefinitionBase[], label: string): P
 async function findTaskByType(allTasks: TaskDefinitionBase[], type: string): Promise<TaskDefinitionBase | undefined> {
     return allTasks.find(t => t.type === type);
 }
+
+/**
+ * Normalizes a platform string or object to a standardized `ContainerPlatform` object.
+ *
+ * @param platform The platform string or object to normalize.
+ * @returns A standardized `ContainerPlatform` object.
+ * @throws An error if the platform string is malformed.
+ */
+export function normalizePlatform(platform: string | ContainerPlatform): ContainerPlatform {
+
+    if (platform && typeof platform === 'string') {
+        const [os, ...architectureParts] = platform.split('/');
+        const architecture = architectureParts.join('/');
+
+        if (!os || !architecture) {
+            throw new Error('Platform string is malformed. It should be in the format "{os}/{architecture}".');
+        }
+
+        return { os, architecture };
+    }
+
+    return platform as ContainerPlatform || {};

```

Comment:
I'd rather return `undefined` than `{}` if the input is `undefined`
```suggestion
    return platform as ContainerPlatform;
```

Reply:
i'm indifferent on this but I think @danegsta preferred `{}`, @dnegstad what are some thoughts?

Reply:
If we do change to what I suggested, we should also change the function return type to `ContainerPlatform | undefined` to be accurate

Reply:
My main reason for wanting `return platform || {};` was that platform is potentially an empty string at this point. I'd also accept `return platform || undefined;` and that may actually work to remove the need for `as ContainerPlatform` on the return type.

Reply:
I like that!

Reply:
sadly we still need the `as ContainerPlatform`